src/Physics/Plugins/ammoJSPlugin.ts

```diff
@@ -219,10 +222,53 @@ export class AmmoJSPlugin implements IPhysicsEnginePlugin {
     }
 
     /**
-     * Update babylon mesh vertices vertices to match physics world object
+     * Update babylon mesh to match physics world object
      * @param impostor imposter to match
      */
     public afterSoftStep(impostor: PhysicsImpostor): void {
+        if (impostor.type === PhysicsImpostor.RopeImpostor) {
+            this.ropeStep(impostor);
+        }
+        else {
+            this.notRopeStep(impostor);
+        }
+    }
+
+    /**
+     * Update babylon mesh vertices vertices to match physics world softbody or cloth
+     * @param impostor imposter to match
+     */
+    public ropeStep(impostor: PhysicsImpostor): void {
+        var bodyVertices = impostor.physicsBody.get_m_nodes();
+        var nbVertices = bodyVertices.size();
+        var node: any;
+        var nodePositions: any;
+        var x, y, z: number;
+        var path: Array<Vector3> = new Array();
+        for (var n = 0; n < nbVertices; n++) {
+            node = bodyVertices.at(n);
+            nodePositions = node.get_m_x();
+            x = nodePositions.x();
+            y = nodePositions.y();
+            z = nodePositions.z();
+            path.push(new Vector3(x, y, z));
+        }
+        var object = impostor.object;
+        var shape = impostor.getParam("shape");
+        if (impostor._isFromLine) {
+            impostor.object = LinesBuilder.CreateLines("lines", {points: path, instance: <LinesMesh>object});
+        }
+        else {
+            impostor.object = ShapeBuilder.ExtrudeShape("ext", {shape: shape, path: path, instance: <Mesh>object});
+        }
+
+    }
+
+    /**
+     * Update babylon mesh vertices vertices to match physics world softbody or cloth
+     * @param impostor imposter to match
+     */
+    public notRopeStep(impostor: PhysicsImpostor): void {

```

Comment:
Could this be made private or do you expect it to be called externally? Is there a better name for this method eg. _softBodyStep?

Reply:
You are right afterSoftStep, ropeStep and notRopeStep can all be private. In the docs I use the term Soft Body to include all three types softbody (3D), cloth(2D) and rope(1D) so the name _softBody would imply rope as well. I have therefore gone for _softbodyOrClothStep  so as to be clearer.